compliance_suite/functions/client.py

```diff
@@ -24,55 +24,86 @@
 class Client():
     """ This class is used to send REST requests to the provided server URL """
 
-    def __init__(self):
-        """ Initialize the Client object"""
+    def __init__(self, service: str, server: str, version: str):
+        """ Initialize the Client object containing the server details
 
+        Args:
+            service: The API service name
+            server: The server URL to send the request
+            version: The version of the deployed API service
+        """
+
+        self.service: str = service
+        self.server: str = server
+        self.version: str = self.parse_version(version)
+        self.request_headers: dict = REQUEST_HEADERS[service]
+        self.endpoint: str = ""
+        self.uri_params: Dict = {}
+        self.query_params: Dict = {}
+        self.operation: str = ""
+        self.request_body: str = ""
+        self.base_url: str = ""
         self.check_cancel = False   # Checks if the Cancel status is to be validated or not
```

Comment:
Now this got actually more confusing, I'm afraid.

What you should do is:
- Initiate a client with only those parameters that you need for multiple instance methods and that will be the same for every request. You don't want to instantiate a new client for every request you send.
- Remove all parameters from the instance methods that you already pass when instantiating the client and refer to them as `self.param`. No param should be passed to both the class constructor and any of the instance methods.
- Obviously define only those params in an instance method that you really need.

Reply:
Actually, maybe it's best if you remove the changes that you did, create an issue for refactoring the class along the lines discussed above and focus _this_ PR on the semantic versioning only.

Reply:
For the above approach, we will need to instantiate the client from the `JobRunner` instead of `TestRunner.` Right now, a new client is created for every job (and request). This way the common parameters like service, URL, and common headers can be defined only once.

Reply:
Sure, I will revert these changes. This will involve a good refactoring of code and can be done later. I will log an issue for this.